kairos-cli/bin/main.rs

```diff
@@ -1,3 +1,46 @@
+mod commands;
+mod common;
+mod crypto;
+mod error;
+
+use clap::Command;
+use commands::{Deposit, Transfer, Withdraw};
+use std::process;
+
+fn cli() -> Command {
+    Command::new("Kairos Client")
+        .about("CLI for interacting with Kairos")
+        .subcommand(Deposit::new_cmd())
+        .subcommand(Transfer::new_cmd())
+        .subcommand(Withdraw::new_cmd())
+}
+
 fn main() {
-    println!("Hello, world!");
+    let arg_matches = cli().get_matches();
+    let (subcommand_name, matches) = arg_matches.subcommand().unwrap_or_else(|| {
+        // No subcommand provided by user.
+        let _ = cli().print_long_help();
+        process::exit(1);
+    });
+
+    let result = match subcommand_name {
+        Deposit::NAME => Deposit::run(matches),
+        Transfer::NAME => Transfer::run(matches),
+        Withdraw::NAME => Withdraw::run(matches),
```

Comment:
If we had a typed CLI nobody would ever forget to handle a subcommand because the compiler would warn the developer that there is a unmatched case

Reply:
This got fixed with refactor: https://github.com/cspr-rad/kairos/pull/17/commits/f54366613e48e8b5f9f5410a26b2b43124965881.

In case of unmatched case, the following error is thrown:

```
error[E0004]: non-exhaustive patterns: `commands::Command::Foo(_)` not covered
```